parser-library/buffer.cpp

```diff
@@ -52,6 +52,22 @@ struct buffer_detail {
 #endif
 };
 
+#ifndef WIN32
```

Comment:
I'd prefer that these be made `inline static`, and placed into an anonymous namespace, and not conditional on the build target (so therefore named to not collide with the ones available on win32). This is a case where the code is there, and so I don't see a good reason for only turning this on for non-win32 builds.